In [74]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt 
import seaborn as sns
from scipy.fftpack import fft, ifft
from sklearn.model_selection import GridSearchCV, train_test_split, cross_val_predict, cross_val_score
from sklearn.feature_selection import SelectKBest, chi2, SelectFromModel
from sklearn.metrics import accuracy_score, f1_score, roc_auc_score, confusion_matrix
from sklearn.svm import SVC, NuSVC, LinearSVC, SVC
from sklearn.neighbors import KNeighborsClassifier
from sklearn.ensemble import RandomForestClassifier
from sklearn.linear_model import LogisticRegression
from sklearn.pipeline import Pipeline

In [ ]:
df = pd.read_csv('DATA_EMG.csv')
df.head()
df.info()

In [ ]:
data = df.to_numpy()
print(data)

In [ ]:
data[:,-1] = [1 if x == 2 else x for x in data[:,-1]]
columns = ['Time', 'Right Rectus Femoris', 'Left Gluteus Maximus', 'Left Gastrocnemius Medialis', 'Left Semitendinosus', 'Left Biceps Femoris Caput Longus', 'Right Vastus Medialis', 'Right Tibialis Anterior', 'Left Gastrocnemius Lateralis', 'Target']
data_df = pd.DataFrame(data, columns=columns)

In [ ]:
data_df.plot(x='Time', y=['Right Rectus Femoris', 'Left Gluteus Maximus', 'Left Gastrocnemius Medialis', 'Left Semitendinosus', 'Left Biceps Femoris Caput Longus', 'Right Vastus Medialis', 'Right Tibialis Anterior', 'Left Gastrocnemius Lateralis'], kind = 'line', figsize = (20,20), subplots=True)
plt.show()

In [ ]:
freq = 1000

if len(data_df)/freq != len(data_df)//freq:
    cant_vent = len(data_df)//freq + 1
else:
    cant_vent = len(data_df)//freq

def sign_mean():
    global df_mean
    ventana = np.zeros((cant_vent, 10))
    Dataset_2=np.array(data_df)
    Dato=np.zeros((cant_vent, 10))
    for i in range (0, cant_vent):
        ventana= Dataset_2[(i * freq):((i * freq) + freq), :]
        Dato[i]=np.mean(ventana, axis=0)
    columns_mean = ['Time (mean)', 'Right Rectus Femoris (Mean)', 'Left Gluteus Maximus(Mean)' , 'Left Gastrocnemius Medialis (Mean)', 'Left Semitendinosus (Mean)', 'Left Biceps Femoris Caput Longus (Mean)', 'Right Vastus Medialis (Mean)', 'Right Tibialis Anterior (Mean)', 'Left Gastrocnemius Lateralis (Mean)', 'Target']
    df_mean = pd.DataFrame(Dato, columns=columns_mean)
    df_mean.pop('Target')
    return (df_mean)

def sign_median():
    global df_median
    ventana = np.zeros((cant_vent, 10))
    Dataset_2=np.array(data_df)
    Dato=np.zeros((cant_vent, 10))
    for i in range (0, cant_vent):
        ventana= Dataset_2[(i * freq):((i * freq) + freq), :]
        Dato[i]=np.median(ventana, axis=0)
    columns_median = ['Time (Median)', 'Right Rectus Femoris (Median)', 'Left Gluteus Maximus(Median)' , 'Left Gastrocnemius Medialis (Median)', 'Left Semitendinosus (Median)', 'Left Biceps Femoris Caput Longus (Median)', 'Right Vastus Medialis (Median)', 'Right Tibialis Anterior (Median)', 'Left Gastrocnemius Lateralis (Median)', 'Target']
    df_median = pd.DataFrame(Dato, columns=columns_median)
    df_median.pop('Target')
    return (df_median)

def sign_std():
    global df_std
    ventana = np.zeros((cant_vent, 10))
    Dataset_2=np.array(data_df)
    Dato=np.zeros((cant_vent, 10))
    for i in range (0, cant_vent):
        ventana= Dataset_2[(i * freq):((i * freq) + freq), :]
        Dato[i]=np.median(ventana, axis=0)
    columns_std = ['Time (std)', 'Right Rectus Femoris (std)', 'Left Gluteus Maximus(std)' , 'Left Gastrocnemius Medialis (std)', 'Left Semitendinosus (std)', 'Left Biceps Femoris Caput Longus (std)', 'Right Vastus Medialis (std)', 'Right Tibialis Anterior (std)', 'Left Gastrocnemius Lateralis (std)', 'Target']
    df_std = pd.DataFrame(Dato, columns=columns_std)
    df_std.pop('Target')
    return (df_std)

def sign_cov():
    global df_cov
    ventana = np.zeros((cant_vent, 10))
    Dataset_2=np.array(data_df)
    Dato=np.zeros((cant_vent, 10))
    for i in range (0, cant_vent):
        ventana= Dataset_2[(i * freq):((i * freq) + freq), :]
        Dato[i]=np.median(ventana, axis=0)
    columns_cov = ['Time (Cov)', 'Right Rectus Femoris (Cov)', 'Left Gluteus Maximus(Cov)' , 'Left Gastrocnemius Medialis (Cov)', 'Left Semitendinosus (Cov)', 'Left Biceps Femoris Caput Longus (Cov)', 'Right Vastus Medialis (Cov)', 'Right Tibialis Anterior (Cov)', 'Left Gastrocnemius Lateralis (Cov)', 'Target']
    df_cov = pd.DataFrame(Dato, columns=columns_cov)
    df_cov.pop('Target')
    return (df_cov)

def sign_var():
    global df_var
    ventana = np.zeros((cant_vent, 10))
    Dataset_2=np.array(data_df)
    Dato=np.zeros((cant_vent, 10))
    for i in range (0, cant_vent):
        ventana= Dataset_2[(i * freq):((i * freq) + freq), :]
        Dato[i]=np.median(ventana, axis=0)
    columns_var = ['Time (var)', 'Right Rectus Femoris (var)', 'Left Gluteus Maximus(var)' , 'Left Gastrocnemius Medialis (var)', 'Left Semitendinosus (var)', 'Left Biceps Femoris Caput Longus (var)', 'Right Vastus Medialis (var)', 'Right Tibialis Anterior (var)', 'Left Gastrocnemius Lateralis (var)', 'Target']
    df_var = pd.DataFrame(Dato, columns=columns_var)
    return (df_var)

sign_mean()
sign_median()
sign_var()
sign_std()
sign_cov()

In [ ]:
df_mean.head()

In [ ]:
df_median.head()

In [ ]:
df_var.head()

In [ ]:
df_std.head()

In [ ]:
df_cov.head()

In [ ]:
data_df.plot(x='Time', y=['Right Rectus Femoris', 'Left Gluteus Maximus', 'Left Gastrocnemius Medialis', 'Left Semitendinosus', 'Left Biceps Femoris Caput Longus', 'Right Vastus Medialis', 'Right Tibialis Anterior', 'Left Gastrocnemius Lateralis'], kind = 'line', figsize = (20, 25), subplots=True, layout=(8,1), xlabel = 'Time (s)',ylabel='Signal amplitude (mV)', title = 'Raw Data')

#plt.show()

df_mean.plot(x = 'Time (mean)', y = ['Right Rectus Femoris (Mean)', 'Left Gluteus Maximus(Mean)' , 'Left Gastrocnemius Medialis (Mean)', 'Left Semitendinosus (Mean)', 'Left Biceps Femoris Caput Longus (Mean)', 'Right Vastus Medialis (Mean)', 'Right Tibialis Anterior (Mean)', 'Left Gastrocnemius Lateralis (Mean)'], kind = 'line', figsize = (20, 25), subplots=True, layout=(8,1), xlabel = 'Mean time (s)', ylabel='Mean Signal Amplitude (mV)', title='Grouped Data with mean values per second')
plt.show()

In [ ]:
data_df.plot(x='Time', y=['Right Rectus Femoris', 'Left Gluteus Maximus', 'Left Gastrocnemius Medialis', 'Left Semitendinosus', 'Left Biceps Femoris Caput Longus', 'Right Vastus Medialis', 'Right Tibialis Anterior', 'Left Gastrocnemius Lateralis'], kind = 'line', figsize = (20, 25), subplots=True, layout=(8,1), xlabel = 'Time (s)',ylabel='Signal amplitude (mV)', title = 'Raw Data')

#plt.show()

df_median.plot(x = 'Time (Median)', y = ['Right Rectus Femoris (Median)', 'Left Gluteus Maximus(Median)' , 'Left Gastrocnemius Medialis (Median)', 'Left Semitendinosus (Median)', 'Left Biceps Femoris Caput Longus (Median)', 'Right Vastus Medialis (Median)', 'Right Tibialis Anterior (Median)', 'Left Gastrocnemius Lateralis (Median)'], kind = 'line', figsize = (20, 25), subplots=True, layout=(8,1), xlabel = 'Mean time (s)', ylabel='Median Signal Amplitude (mV)', title='Grouped Data with median values per second')
plt.show()

In [ ]:
data_df.plot(x='Time', y=['Right Rectus Femoris', 'Left Gluteus Maximus', 'Left Gastrocnemius Medialis', 'Left Semitendinosus', 'Left Biceps Femoris Caput Longus', 'Right Vastus Medialis', 'Right Tibialis Anterior', 'Left Gastrocnemius Lateralis'], kind = 'line', figsize = (20, 25), subplots=True, layout=(8,1), xlabel = 'Time (s)',ylabel='Signal amplitude (mV)', title = 'Raw Data')

#plt.show()

df_std.plot(x = 'Time (std)', y = ['Right Rectus Femoris (std)', 'Left Gluteus Maximus(std)' , 'Left Gastrocnemius Medialis (std)', 'Left Semitendinosus (std)', 'Left Biceps Femoris Caput Longus (std)', 'Right Vastus Medialis (std)', 'Right Tibialis Anterior (std)', 'Left Gastrocnemius Lateralis (std)'], kind = 'line', figsize = (20, 25), subplots=True, layout=(8,1), xlabel = 'std time (s)', ylabel='std Signal Amplitude (mV)', title='Grouped Data with std values per second')
plt.show()

In [ ]:
data_df.plot(x='Time', y=['Right Rectus Femoris', 'Left Gluteus Maximus', 'Left Gastrocnemius Medialis', 'Left Semitendinosus', 'Left Biceps Femoris Caput Longus', 'Right Vastus Medialis', 'Right Tibialis Anterior', 'Left Gastrocnemius Lateralis'], kind = 'line', figsize = (20, 25), subplots=True, layout=(8,1), xlabel = 'Time (s)',ylabel='Signal amplitude (mV)', title = 'Raw Data')

#plt.show()

df_cov.plot(x = 'Time (Cov)', y = ['Right Rectus Femoris (Cov)', 'Left Gluteus Maximus(Cov)' , 'Left Gastrocnemius Medialis (Cov)', 'Left Semitendinosus (Cov)', 'Left Biceps Femoris Caput Longus (Cov)', 'Right Vastus Medialis (Cov)', 'Right Tibialis Anterior (Cov)', 'Left Gastrocnemius Lateralis (Cov)'], kind = 'line', figsize = (20, 25), subplots=True, layout=(8,1), xlabel = 'Mean time (s)', ylabel='Cov Signal Amplitude (mV)', title='Grouped Data with Cov values per second')
plt.show()

In [ ]:
data_df.plot(x='Time', y=['Right Rectus Femoris', 'Left Gluteus Maximus', 'Left Gastrocnemius Medialis', 'Left Semitendinosus', 'Left Biceps Femoris Caput Longus', 'Right Vastus Medialis', 'Right Tibialis Anterior', 'Left Gastrocnemius Lateralis'], kind = 'line', figsize = (20, 25), subplots=True, layout=(8,1), xlabel = 'Time (s)',ylabel='Signal amplitude (mV)', title = 'Raw Data')

#plt.show()

df_var.plot(x = 'Time (var)', y = ['Right Rectus Femoris (var)', 'Left Gluteus Maximus(var)' , 'Left Gastrocnemius Medialis (var)', 'Left Semitendinosus (var)', 'Left Biceps Femoris Caput Longus (var)', 'Right Vastus Medialis (var)', 'Right Tibialis Anterior (var)', 'Left Gastrocnemius Lateralis (var)'], kind = 'line', figsize = (20, 25), subplots=True, layout=(8,1), xlabel = 'Mean time (s)', ylabel='var Signal Amplitude (mV)', title='Grouped Data with var values per second')
plt.show()

#Frequency Analysis with Fourier Transform

# Concatenating all dataframes

In [86]:
stats_df = pd.concat([df_mean, df_median, df_std, df_cov, df_var], axis=1)
stats_df

Time (mean)  Right Rectus Femoris (Mean)  Left Gluteus Maximus(Mean)  \
0         0.4995                     0.000045                1.923540e-05   
1         1.4995                    -0.000045               -1.810325e-06   
2         2.4995                     0.000136                1.489724e-06   
3         3.4995                     0.000080                1.984335e-06   
4         4.4995                    -0.000375               -2.246663e-05   
..           ...                          ...                         ...   
682     682.4995                     0.000003                2.906505e-05   
683     683.4995                    -0.000034                5.821844e-07   
684     684.4995                     0.000063                2.018275e-05   
685     685.4995                    -0.000028               -4.521857e-05   
686     686.2620                     0.000018                7.036150e-06   

     Left Gastrocnemius Medialis (Mean)  Left Semitendinosus (Mean)  \
0                             -0.000080                    0.000058   
1                              0.000053                   -0.000014   
2                              0.000276                   -0.000027   
3                             -0.000374                   -0.000012   
4                              0.000177                    0.000042   
..                                  ...                         ...   
682                            0.000350                    0.000132   
683                            0.000015                   -0.000226   
684                           -0.001522                    0.000203   
685                            0.000508                   -0.000068   
686                            0.001705                   -0.000159   

     Left Biceps Femoris Caput Longus (Mean)  Right Vastus Medialis (Mean)  \
0                                   0.000043                     -0.000021   
1                                   0.000021                      0.000021   
2                                  -0.000051                     -0.000115   
3                                  -0.000050                     -0.000027   
4                                   0.000037                      0.000161   
..                                       ...                           ...   
682                                -0.000010                      0.000235   
683                                -0.000065                      0.000332   
684                                 0.000192                     -0.000427   
685                                -0.000050                     -0.000071   
686                                -0.000094                      0.000076   

     Right Tibialis Anterior (Mean)  Left Gastrocnemius Lateralis (Mean)  \
0                     -2.840982e-05                             0.000211   
1                     -6.771324e-05                            -0.000340   
2                      5.225433e-05                             0.000093   
3                     -6.911869e-05                            -0.000227   
4                      1.295356e-04                            -0.000022   
..                              ...                                  ...   
682                   -2.749086e-07                            -0.000255   
683                   -1.195577e-04                             0.000593   
684                    8.383121e-05                            -0.000648   
685                   -4.443207e-05                             0.000172   
686                    9.654445e-05                            -0.000271   

     Time (Median)  ...  Time (var)  Right Rectus Femoris (var)  \
0           0.4995  ...      0.4995                    0.000211   
1           1.4995  ...      1.4995                   -0.000923   
2           2.4995  ...      2.4995                    0.000710   
3           3.4995  ...      3.4995                    0.000738   
4           4.4995  ...      4.4995       

In [ ]:
print(stats_df.columns)

In [60]:
time_array = stats_df[['Time (Median)']]
stats_df.drop(['Time (mean)', 'Time (Median)', 'Time (std)', 'Time (Cov)', 'Time (var)'], axis=1, inplace = True)

In [65]:
stats_df.columns

Index(['Right Rectus Femoris (Mean)', 'Left Gluteus Maximus(Mean)',
       'Left Gastrocnemius Medialis (Mean)', 'Left Semitendinosus (Mean)',
       'Left Biceps Femoris Caput Longus (Mean)',
       'Right Vastus Medialis (Mean)', 'Right Tibialis Anterior (Mean)',
       'Left Gastrocnemius Lateralis (Mean)', 'Right Rectus Femoris (Median)',
       'Left Gluteus Maximus(Median)', 'Left Gastrocnemius Medialis (Median)',
       'Left Semitendinosus (Median)',
       'Left Biceps Femoris Caput Longus (Median)',
       'Right Vastus Medialis (Median)', 'Right Tibialis Anterior (Median)',
       'Left Gastrocnemius Lateralis (Median)', 'Right Rectus Femoris (std)',
       'Left Gluteus Maximus(std)', 'Left Gastrocnemius Medialis (std)',
       'Left Semitendinosus (std)', 'Left Biceps Femoris Caput Longus (std)',
       'Right Vastus Medialis (std)', 'Right Tibialis Anterior (std)',
       'Left Gastrocnemius Lateralis (std)', 'Right Rectus Femoris (Cov)',
       'Left Gluteus Maximus(Cov)

In [ ]:
time_array.head()

# Exploratory review of the Transformed DataFrame

In [ ]:
stats_df.info()
stats_df.describe()

# Parameter Selection through Feature Engineering techniques

In [ ]:
X = stats_df.iloc[:, :-1]
y = stats_df.iloc[:, -1]

model = RandomForestClassifier()
model.fit(X,y)
feat_importances = pd.Series(model.feature_importances_, index=X.columns)
feat_importances.nlargest(10).plot(kind='barh')
plt.show()
print(feat_importances.nlargest(10))

In [ ]:
# Matrix correlation:
plt.figure(figsize=(50,50))
sns.heatmap(data = stats_df.corr(),annot=True,cmap="mako")
plt.show()


In [ ]:
#Splitting the dataset into the Training set and Test set

X_train, X_test, y_train, y_test = train_test_split(stats_df[['Left Gastrocnemius Medialis (Mean)', 'Left Gastrocnemius Medialis (Median)','Left Gastrocnemius Medialis (Cov)', 'Left Gastrocnemius Medialis (var)', 'Left Biceps Femoris Caput Longus (Median)', 'Left Biceps Femoris Caput Longus (var)', 'Left Biceps Femoris Caput Longus (Cov)']], y, test_size = 0.85, random_state = 85, stratify=y)

#Let's check the shape of the training and test sets.

print(X_train.shape, y_train.shape, X_test.shape, y_test.shape)

# Fare review of classification methods for EMG Dataset

In [ ]:
#Create a function that fit the model in a Random Forest Classifier and return the accuracy score.

def random_forest(X_train, y_train, X_test, y_test):
    global y_pred, best_trees, best_accuracy, best_criterion
    tree_num = []
    accuracy = []
    criterion = []
    for i in range(1, 109, 9):
        for j in ['gini', 'entropy']:
            rf = RandomForestClassifier(n_estimators=i, random_state=42, criterion=j, n_jobs=4)
            rf.fit(X_train, y_train)
            y_pred = rf.predict(X_test)
            y_pred_train = rf.predict(X_train)
            a_s_test= accuracy_score(y_test, y_pred)
            a_s_train = accuracy_score(y_train, y_pred_train)
            print(f'Accuracy score test for {i} trees and criterion {j}: {a_s_test}')
            print(f'Accuracy score train for {i} trees and criterion {j}: {a_s_train}')
            tree_num.append(i)
            accuracy.append(accuracy_score(y_test, y_pred))
            #Save the best accuracy_score and the number of trees.3
            if i == 1:
                best_accuracy = a_s_test
                best_accuracy_train = a_s_train
                best_trees = i
                best_criterion = j
            elif accuracy_score(y_test, y_pred) > best_accuracy:
                best_accuracy = accuracy_score(y_test, y_pred)
                best_accuracy_train = accuracy_score(y_train, y_pred_train)
                best_trees = i
                best_criterion = j
    plt.figure(figsize=(20,10))
    plt.plot(tree_num, accuracy)
    plt.xlabel('Number of trees')
    plt.ylabel('Accuracy')
    plt.show()

    return f'The best accuracy score for test subset is: {best_accuracy} and the best for train subset is: {best_accuracy_train} with number of trees: {best_trees} and criterion {best_criterion}'

random_forest(X_train, y_train, X_test, y_test)

In [ ]:
def knn(X_train, y_train, X_test, y_test):
    global y_pred, best_k, best_accuracy, best_weight, best_p
    k = []
    accuracy = []
    for i in range(1, 20, 1):
        for j in ['uniform', 'distance']:
            for power in [1,2,3,4,5]:
                knn = KNeighborsClassifier(n_neighbors=i, n_jobs=-1, weights=j, p=power)
                knn.fit(X_train, y_train)
                y_pred = knn.predict(X_test)
                y_pred_train = knn.predict(X_train)
                a_s_test= accuracy_score(y_test, y_pred)
                a_s_train = accuracy_score(y_train, y_pred_train)
                print(f'Accuracy score for {i} neighbors in test: {a_s_test} and for {i} neighbors in train: {a_s_train} weight {j} with a power of {power}')
                k.append(i)
                accuracy.append(accuracy_score(y_test, y_pred))
                #Save the best accuracy_score and the number of neighbors.
                if i == 1:
                    best_accuracy = a_s_test
                    best_accuracy_train = a_s_train
                    best_k = i
                    best_weight = j
                    best_p=power
                elif accuracy_score(y_test, y_pred) > best_accuracy:
                    best_accuracy = a_s_test
                    best_accuracy_train = a_s_train
                    best_k = i
                    best_weight = j
                    best_p = power
    
    plt.figure(figsize=(20,10))
    plt.plot(k, accuracy)
    plt.plot()
    plt.xlabel('Number of neighbors')
    plt.ylabel('Accuracy')
    plt.show()

    return f'The best accuracy score for test is: {best_accuracy}, for train is: {best_accuracy_train} and the best number of neighbors is: {best_k} with weight defined as {best_weight} with a power of {best_p}'

In [ ]:
knn(X_train, y_train, X_test, y_test)

In [ ]:
def gamma_SVM(X_train, y_train, X_test, y_test):
    gamma = []
    accuracy = []
    for i in np.arange(0.1, 3, 0.5):
        for j in [0.1, 1, 10, 100]:
            model = SVC(random_state=None, kernel='rbf', C=j, gamma=i)
            model.fit(X_train, y_train)
            y_pred = model.predict(X_test)
            print(f'Accuracy score for gamma {i} and C {j}: {accuracy_score(y_test, y_pred)}')
            gamma.append(i)
            accuracy.append(accuracy_score(y_test, y_pred))
    plt.figure(figsize=(20,10))
    plt.plot(gamma, accuracy)
    plt.xlabel('Gamma')
    plt.ylabel('Accuracy')
    plt.show()

    return f'The best accuracy score is: {max(accuracy)} and the best gamma is: {gamma[accuracy.index(max(accuracy))]} with C {j}'

In [ ]:
gamma_SVM(X_train, y_train, X_test, y_test)

# Grid Search application to classifiers

In [ ]:
pipe = Pipeline([("classifier", KNeighborsClassifier())])
search_space = [
    {
        "classifier" : [LogisticRegression()],
        "classifier__penalty" : ['none', 'l1', 'l2', 'elasticnet'],
        "classifier__max_iter" : [i for i in range(1, 10000, 1000)],
        "classifier__n_jobs" : [1, 2, 3, 4, -1]
     },

     {
        "classifier" : [RandomForestClassifier()],
        "classifier__n_estimators" : [i for i in range(1,1000,100)],
        "classifier__criterion" : ['gini', 'entropy', 'log_loss'],
        "classifier__class_weight" : ['balanced', 'balanced_subsample']   
     },

     {
        "classifier" : [KNeighborsClassifier()],
        "classifier__n_neighbors" : [i for i in range(1, 20, 1)],
        "classifier__weights" : ['uniform', 'distance'],
        "classifier__algorithm" : ['auto', 'ball_tree', 'kd_tree', 'brute'],
        "classifier__p" : [1, 2]
     },

     {
        "classifier" : [NuSVC()],
        "classifier__nu" : [0.2, 0.4, 0.6, 0.8, 1],
        "classifier__kernel" : ['linear', 'poly', 'rbf', 'sigmoid'],
        "classifier__decision_function_shape" : ['ovo', 'ovr']
     },

     {
        "classifier" : [LinearSVC()],
        "classifier__penalty" : ['l1', 'l2'],
        "classifier__loss" : ['hinge', 'squared_hinge'],
        "classifier__multi_class" : ['ovr', 'crammer_singer'],
        "classifier__max_iter" : [i for i in range(1, 5000, 200)]
     },

     {
        "classifier" : [SVC()],
        "classifier__kernel" : ['linear', 'poly', 'rbf', 'sigmoid'],
        "classifier__gamma" : ['scale', 'auto']
     }
]

In [ ]:
#GridSearch:

classif = GridSearchCV(pipe, search_space, cv=15, verbose=0, n_jobs=-1)
best_clf=classif.fit(X_train, y_train)
print(f'El mejor modelo es: {best_clf.best_estimator_.get_params()["classifier"]}, con un accuracy de: {classif.score(X_train, y_train)}')

In [ ]:
x_features =stats_df[['Left Gastrocnemius Medialis (Mean)', 'Left Gastrocnemius Medialis (Median)','Left Gastrocnemius Medialis (Cov)', 'Left Gastrocnemius Medialis (var)', 'Left Biceps Femoris Caput Longus (Median)', 'Left Biceps Femoris Caput Longus (var)', 'Left Biceps Femoris Caput Longus (Cov)']]

# Confusion Matrix plot

In [ ]:
y_pred = best_clf.best_estimator_.predict(x_features)
cf_mtx = confusion_matrix(y, y_pred)
ax = sns.heatmap(cf_mtx, annot=True, cmap= 'mako', xticklabels= ['Positive', 'Negative'],
            yticklabels = ['Positive', 'Negative'], fmt='g')
ax.set(xlabel = "Actual Values", ylabel="Predicted Values")

De los 686 datos almacenados, 559 fueron predichos correctamente cómo verdaderos positivos y verdaderos negativos

# Metric analysis

In [ ]:
print(accuracy_score(y, y_pred), f1_score(y, y_pred), roc_auc_score(y, y_pred))
